homeassistant/components/husqvarna_automower/sensor.py

```diff
@@ -0,0 +1,169 @@
+"""Creates a the sensor entities for the mower."""
+from collections.abc import Callable
+from dataclasses import dataclass
+import datetime
+import logging
+
+from aioautomower.model import MowerAttributes, MowerModes
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory, UnitOfLength, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AutomowerDataUpdateCoordinator
+from .entity import AutomowerBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class AutomowerSensorEntityDescription(SensorEntityDescription):
+    """Describes Automower sensor entity."""
+
+    exists_fn: Callable[[MowerAttributes], bool] = lambda _: True
+    value_fn: Callable[[MowerAttributes], str]
+
+
+SENSOR_TYPES: tuple[AutomowerSensorEntityDescription, ...] = (
+    AutomowerSensorEntityDescription(
+        key="battery_percent",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda data: data.battery.battery_percent,
+    ),
+    AutomowerSensorEntityDescription(
+        key="mode",
+        translation_key="mode",
+        device_class=SensorDeviceClass.ENUM,
+        options=[option.lower() for option in list(MowerModes)],
+        value_fn=lambda data: data.mower.mode.lower()
+        if data.mower.mode != MowerModes.UNKNOWN
+        else None,
```

Comment:
Please use a parenthesis around the lambda and indent it so it's easier to distinguish the entity description parameters from the lambda.